### PR TITLE
Rating system for outgoing messages introduced

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

File: libp2p/Session.cpp [hunk 1]
```diff
@@ -382,7 +382,7 @@ void Session::doRead()
 		}
 		catch (std::exception const& _e)
 		{
-			clog(NetWarn) << "Exception decoding frame header RLP:" << bytesConstRef(m_data.data(), h128::size).cropped(3);
+			clog(NetWarn) << _e.what() << "Exception decoding frame header RLP:" << bytesConstRef(m_data.data(), h128::size).cropped(3);
```
Comment (subtly) [inline libp2p/Session.cpp]: Unrelated... was an exception hit here?

File: libwhisper/WhisperHost.cpp [hunk 2]
```diff
@@ -66,21 +68,31 @@ void WhisperHost::inject(Envelope const& _m, WhisperPeer* _p)
...
-	{
-		for (auto const& f: m_filters)
-			if (f.second.filter.matches(_m))
-				for (auto& i: m_watches)
-					if (i.second.id == f.first)
-						i.second.changes.push_back(h);
-	}
+	int rating = 1; // rating for local host is based upon: 1. installed watch; 2. proof of work
+
+	//if (bloomfilter.check)
```
Comment (subtly) [inline libwhisper/WhisperHost.cpp]: Remove commented code?

File: libwhisper/WhisperHost.cpp [hunk 3]
```diff
@@ -66,21 +68,31 @@ void WhisperHost::inject(Envelope const& _m, WhisperPeer* _p)
...
+	//{
+		// rating *= 10;
+		if (_p) // originated externally
+			DEV_GUARDED(m_filterLock)
+			{
```
Comment (subtly) [inline libwhisper/WhisperHost.cpp]: curlies

File: libwhisper/WhisperHost.cpp [hunk 4]
```diff
@@ -66,21 +68,31 @@ void WhisperHost::inject(Envelope const& _m, WhisperPeer* _p)
...
+				for (auto const& f: m_filters)
+					if (f.second.filter.matches(_m))
+						for (auto& i: m_watches)
+							if (i.second.id == f.first)
+							{
+								i.second.changes.push_back(h);
+								rating *= 10;
```
Comment (subtly) [inline libwhisper/WhisperHost.cpp]: += 10
The nth message isn't exponentially more relevant than a previous messages.

File: libp2p/Session.cpp (hunk 1 above)
Comment (gluk256) [inline libp2p/Session.cpp]: Warning: unused variable. 
Changing this line...

**Comment by gluk256** [comment]
fixed

**Pull request merged**